### ISSUE TITLE
@nuxt/test-utils is not working with useI18n from @nuxtjs/i18n

### ISSUE THREAD

**Issue by trandaison** [nuxt/test-utils#743]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.16.0`
- Nuxt Version:     `3.9.3`
- CLI Version:      `3.10.0`
- Nitro Version:    `2.8.1`
- Package Manager:  `pnpm@8.15.0`
- Builder:          `-`
- User Config:      `devtools`, `modules`
- Runtime Modules:  `@nuxtjs/i18n@8.0.0`
- Build Modules:    `-`

### Reproduction

https://github.com/trandaison/test-utils/tree/main/examples/i18n

### Describe the bug

I forked these [examples](https://github.com/nuxt/test-utils/tree/main/examples/i18n) then add a component named `TheHeader.vue` which do use `const { t } = useI18n()` in the setup script.
The test will be timeout and  unable to run in this case.

![Screenshot 2024-01-29 at 17 18 18](https://github.com/nuxt/test-utils/assets/13447235/2265a089-a5cf-490f-8f7a-a7e1924daa66)

If I remove the `useI18n` hook and replace with `$t` or don't use any translate function at all, the test will be passed.

**Comment by darthf1** [comment]
Does it also fail on [v3.10.0](https://github.com/nuxt/test-utils/releases/tag/v3.10.0)? I'm hitting (i think) similar errors when updating to `v3.11.0`, but I still need to investigate.

**Comment by trandaison** [comment]
@darthf1 I confirm that the error happens on v3.10.0 as well.

**Comment by StevenPewsey** [comment]
I have found that the below solution works

```
<script setup lang="ts">
const { t } = useNuxtApp().$i18n
</script>
```

**Comment by trandaison** [comment]
@StevenPewsey Yes, it works as same as `$t`, because of avoid using the `useI18n` composition.
`useI18n()` didn't work, maybe some other compositions from other 3rd plugin have the same problem, I guess.

**Comment by indiehjaerta** [comment]
Is this beeing looked at?  It's currently holding our enterprise back from upgrading.

Still present using the latest version 
    "nuxt": "^3.11.2",
    "@nuxt/test-utils": "^3.12.0",
    "@nuxtjs/i18n": "^8.3.0",

**Comment by trandaison** [comment]
@indiehjaerta Here is my workaround so far.
I create a test setup like this: 

```ts
import { beforeAll, afterAll, vi } from 'vitest';
import { config } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';

beforeAll(() => {
  vi.mock('vue-i18n', async (importOriginal) => {
    const module = await importOriginal<typeof import('vue-i18n')>();

    return { ...module };
  });
});

afterAll(() => {
  vi.restoreAllMocks();
});

// setup i18n
const i18n = createI18n({
  legacy: false,
  locale: 'en',
  locales: [{ code: 'en', iso: 'en-US', file: 'en-US.yml', name: 'English' }],
  messages: {},
  missing: (locale: string, key: string) => `trans:__${key}__`,
});

config.global.plugins.push(i18n);
```

**Comment by ileskaa** [comment]
> I have found that the below solution works
> 
> ```
> <script setup lang="ts">
> const { t } = useNuxtApp().$i18n
> </script>
> ```

Amazing, thanks!

**Comment by rigtigeEmil** [comment]
Is there still no solution for this, apart from having to modify the way `t` is injected in scripts?

**Comment by ThomasBerneHCSE** [comment]
Found a better way to add it in a setup file.
```ts
import { mockNuxtImport } from '@nuxt/test-utils/runtime';

mockNuxtImport('useI18n', () => {
  return () => useNuxtApp().$i18n;
});
```

**Comment by derHodrig** [comment]
> Found a better way to add it in a setup file.
> 
> import { mockNuxtImport } from '@nuxt/test-utils/runtime';
> 
> mockNuxtImport('useI18n', () => () => useNuxtApp().$i18n);

This does not work in my case "Error: Cannot find import "useI18n" to mock"
my tests are under "projectRoot/tests/unit/*

```
export default defineVitestConfig({
  resolve: {
    alias: {
      '@layerComponents': path.resolve(__dirname, 'components'),
    },
  },
  test: {
    environment: 'nuxt',
    environmentOptions: {
      nuxt: {
        rootDir: fileURLToPath(new URL('./playground', import.meta.url)),
        domEnvironment: 'happy-dom',
        mock: {
          intersectionObserver: true,
          indexedDb: true,
        },
      },
    },
    include: ['tests/unit/**/*.spec.ts'],
  },
})
```

The console also prompts a number of available "imports" and useI18n is not there but at .playground/.nuxt/imports.d.ts there it is. 
Testing is such a hassle

**Comment by IT-WIBRC** [comment]
```ts
import { vi } from "vitest";
import { ref } from  "vue";
import { mockNuxtImport } from "@nuxt/test-utils/runtime";

const mockI18n = {
  useI18n: () => ({
    t: vi.fn((msg: string) => msg),
    locale: ref("en"),
    locales: [
      { code: "en",
        name: "English" },
      { code: "fr",
        name: "French" },
    ],
    setLocale: vi.fn((locale: string) => {
      mockI18n.useI18n().locale.value = locale;
    }),
    $t: (key: string) => `Mock $t: ${key}`,
    $i18n: {
      locale: "en",
      locales: [
        { code: "en",
          name: "English" },
        { code: "fr",
          name: "French" },
      ],
    },
  }),
};

mockNuxtImport("useI18n", () => mockI18n.useI18n);
```

You can also use `vitest.setup.ts` file when you paste this. Also add it inside the `setupFiles` property in your `vitest.config.ts`.

**Comment by ThomasBerneHCSE** [comment]
@danielroe just upgraded to the new version (@nuxt/test-utils@3.23.0) including @yamachi4416 changes. I get this new error.

I removed the changes I posted on this issue (https://github.com/nuxt/test-utils/issues/743#issuecomment-2501333640) and got the same error. What is the good practice to make nuxt i18n and nuxt test utils work together.

<img width="1473" height="433" alt="Image" src="https://github.com/user-attachments/assets/128ae713-d028-407e-9a68-e63a7210aea6" />

**Comment by danielroe** [comment]
@ThomasBerneHCSE do you have a reproduction?

**Comment by ThomasBerneHCSE** [comment]
I'll try to make one today. Give me some time.

**Comment by danielroe** [comment]
of course! 😊

**Comment by ThomasBerneHCSE** [comment]
@danielroe I investigated my issue. It looks like it's a @nuxtjs/color-mode 4.0.0 issue. With the 3.x.x version of @nuxtjs/color-mode everything works fine but when I upgrade it to v4 it breaks all the tests.
Here is a reproduction :
https://github.com/ThomasBerneHCSE/nuxt-i18n-on-test-reproduction.

I'll post an issue on the @nuxtjs/color-mode repository.

**Comment by ThomasBerneHCSE** [comment]
https://github.com/nuxt-modules/color-mode/issues/342
